rpcs3/Emu/Io/GHLtar.cpp

```diff
@@ -0,0 +1,204 @@
+﻿#include "stdafx.h"
+#include "GHLtar.h"
+#include "Emu/Cell/lv2/sys_usbd.h"
+#include "Input/pad_thread.h"
+#include <Emu\Cell\Modules\cellPad.cpp>
+#include <Emu\RSX\GL\GLHelpers.h>
+
+LOG_CHANNEL(ghltar_log);
+
+usb_device_ghltar::usb_device_ghltar()
+{
+	device        = UsbDescriptorNode(USB_DESCRIPTOR_DEVICE, UsbDeviceDescriptor{0x0200, 0x00, 0x00, 0x00, 0x20, 0x12BA, 0x074B, 0x0100, 0x01, 0x02, 0x00, 0x01});
+	auto& config0 = device.add_node(UsbDescriptorNode(USB_DESCRIPTOR_CONFIG, UsbDeviceConfiguration{0x0029, 0x01, 0x01, 0x00, 0x80, 0x96}));
+	config0.add_node(UsbDescriptorNode(USB_DESCRIPTOR_INTERFACE, UsbDeviceInterface{0x00, 0x00, 0x02, 0x03, 0x00, 0x00, 0x00}));
+	config0.add_node(UsbDescriptorNode(USB_DESCRIPTOR_HID, UsbDeviceHID{0x0111, 0x00, 0x01, 0x22, 0x001d}));
+	config0.add_node(UsbDescriptorNode(USB_DESCRIPTOR_ENDPOINT, UsbDeviceEndpoint{0x81, 0x03, 0x0020, 0x01}));
+	config0.add_node(UsbDescriptorNode(USB_DESCRIPTOR_ENDPOINT, UsbDeviceEndpoint{0x01, 0x03, 0x0020, 0x01}));
+}
+
+usb_device_ghltar::~usb_device_ghltar()
+{
+}
+
+void usb_device_ghltar::control_transfer(u8 bmRequestType, u8 bRequest, u16 wValue, u16 wIndex, u16 wLength, u32 buf_size, u8* buf, UsbTransfer* transfer)
+{
+	transfer->fake = true;
+
+	// Control transfers are nearly instant
+	switch (bmRequestType)
```

Comment:
I don't like how all of the code inside of this is basically pressed into one liners